File: backend/api/src/main/java/edu/miu/cs425/flightbookingsystem/dto/FlightScheduleDTO.java
package edu.miu.cs425.flightbookingsystem.dto;


import java.math.BigDecimal;
import java.time.LocalDate;

public record FlightScheduleDTO(Long id,
                                LocalDate departureDate,
                                Integer availableSeats,
                                Integer maximumSeats,
                                BigDecimal basePrice,
                                FlightRouteDTO flightRouteDTO
) {
}

File: frontend/flight-booking/src/model/FlightScheduleDTO.ts
import { FlightRouteDTO } from './FlightRouteDTO';

export interface FlightScheduleDTO {
  id?: number;
  departureDate: Date;
  availableSeats: number;
  maximumSeats: number;
  basePrice: number;
  flightRouteDTO: FlightRouteDTO;
}
